barrier.go

```diff
@@ -158,9 +158,7 @@ func (n *idleBarrier) Point(m edge.PointMessage) (edge.Message, error) {
 }
 
 func (n *idleBarrier) resetTimer() {
-	if n.idle > 0 {
-		n.timer.Reset(n.idle)
-	}
+	n.timer.Reset(n.idle)
```

Comment:
How does this change prevent the race on resetting the timer?

Reply:
the n.idle > 0 check is an unnecessary carry-over from when the idle and periodic barrier were one implementation, since idleBarrier wouldn't be instantiated unless n.idle is > 0 this is an unnecessary check..

also potential race conditions on resetting the timer doesn't seem necessary to check for, timer.Reset will advance the expiration to now + duration and since we're always advancing/resetting by n.idle any minute difference between two simultaneous writers shouldn't have much of an impact

Reply:
OK, removing the check makes sense. 

The issue with the race is that it can cause undefined state in the timer, thus rendering the timer useless. It may even cause a panic depending on how the timer is implemented. To safely reuse a timer we have to protect the Reset process as documented here https://golang.org/pkg/time/#Timer.Reset.